airflow/models/dagrun.py

```diff
@@ -94,8 +103,14 @@ def __init__(
         self.conf = conf or {}
         self.state = state
         self.run_type = run_type
+        self.dag_hash = dag_hash
+        self.callback: Optional[callback_requests.DagCallbackRequest] = None
```

Comment:
Can we return the callback as a result of the update_state method rather than setting the property on the DB object? It sounds weird to add non-DB property in the DB model.

Reply:
So have `ready_tis, callbacks_to_run = dagrun.update_state()`?

Reply:
I think that's the easiest way - if we add type checking it's not even bad (and actually maybe we can also find a better name for the update_state method which we wrote about elsewhere. The update_state is not a really good name now as this method does more than that.

Reply:
removed `dagrun.callback` in https://github.com/apache/airflow/pull/10956/commits/d8d115469c1d4436a7a1e9463df5c4c5fda2e26d

Will defer the name changing of update_state to a later PR since we haven't changed any behavior i.e callbacks were executed previously. We just added a flag which will not execute the callbacks there.